tests/component/encapsulation.cpp

```diff
@@ -0,0 +1,31 @@
+#include "gtest/gtest.h"
+
+#include <libcellml>
+
+
+TEST(Encapsulation, serialise) {
+    libcellml::Component parent = libcellml::Component();
+    libcellml::Component child = libcellml::Component();
+    parent.addComponent(child);
```

Comment:
I don't think the above is tenable unless you change to shared_ptr as I expect the internal representation of a list of children to be an STL container of shared_ptr which can't be generated except from another shared_ptr.